src/nupic/algorithms/TemporalMemory.cpp

```diff
@@ -1084,50 +1068,3 @@ bool TemporalMemory::operator==(const TemporalMemory &other) {
 bool TemporalMemory::operator!=(const TemporalMemory &other) {
   return !(*this == other);
 }
-
-//----------------------------------------------------------------------
-// Debugging helpers
-//----------------------------------------------------------------------
-
-// Print the main TM creation parameters
-void TemporalMemory::printParameters() {
-  std::cout << "------------CPP TemporalMemory Parameters ------------------\n";
-  std::cout
-      << "version                   = " << TM_VERSION << std::endl
-      << "numColumns                = " << numberOfColumns() << std::endl
-      << "cellsPerColumn            = " << getCellsPerColumn() << std::endl
-      << "activationThreshold       = " << getActivationThreshold() << std::endl
-      << "initialPermanence         = " << getInitialPermanence() << std::endl
-      << "connectedPermanence       = " << getConnectedPermanence() << std::endl
-      << "minThreshold              = " << getMinThreshold() << std::endl
-      << "maxNewSynapseCount        = " << getMaxNewSynapseCount() << std::endl
-      << "permanenceIncrement       = " << getPermanenceIncrement() << std::endl
-      << "permanenceDecrement       = " << getPermanenceDecrement() << std::endl
-      << "predictedSegmentDecrement = " << getPredictedSegmentDecrement()
-      << std::endl
-      << "maxSegmentsPerCell        = " << getMaxSegmentsPerCell() << std::endl
-      << "maxSynapsesPerSegment     = " << getMaxSynapsesPerSegment()
-      << std::endl;
-}
-
-void TemporalMemory::printState(vector<UInt> &state) {
-  std::cout << "[  ";
-  for (UInt i = 0; i != state.size(); ++i) {
-    if (i > 0 && i % 10 == 0) {
-      std::cout << "\n   ";
-    }
-    std::cout << state[i] << " ";
-  }
-  std::cout << "]\n";
-}
-
-void TemporalMemory::printState(vector<Real> &state) {
-  std::cout << "[  ";
-  for (UInt i = 0; i != state.size(); ++i) {
-    if (i > 0 && i % 10 == 0) {
-      std::cout << "\n   ";
-    }
-    std::printf("%6.3f ", state[i]);
-  }
-  std::cout << "]\n";
-}
```

Comment:
These Print functions might be useful for debugging but if every class has a human readable serialization then we should not need it.  So, ok we can do without these.